config/initializers/omniauth.rb

```diff
@@ -3,7 +3,23 @@
 OmniAuth.config.logger = Rails.logger
 
 Rails.application.config.middleware.use OmniAuth::Builder do
-  provider :google_oauth2, ENV['GOOGLE_CONSUMER_KEY'], ENV['GOOGLE_CONSUMER_SECRET'], verify_iss: false
+  provider :google_oauth2, {
+    verify_iss: false,
+    setup: proc do |env|
+      host = env['SERVER_NAME']
```

Comment:
might be a good idea to move the whole subdomain scrabble into some kind of helper since it's necessary for github tokens as well.

it might be worth pulling stuff into the `Usergroup` class and move initialization to a middleware so you dont have to do the subdomain handling twice (see `WhitelabelDetection`).

Reply:
extraction is done in https://github.com/rughh/on_ruby/pull/1108